simavr/sim/avr_timer.c

```diff
@@ -159,6 +159,117 @@ avr_timer_compc(
 	return avr_timer_comp((avr_timer_t*)param, when, AVR_TIMER_COMPC);
 }
 
+/**
+ * @brief avr_timer_irq_ext_clock - External clock event handler.
+ * All the activity originates here if timer is clocked via its external pin Tn.
+ * Called on every change of the Tn pin (should be connected to its IRQ via avr_irq_register_notify).
+ * External events are unpredictable, so we cannot schedule overflows and compare matches
+ * as we do with system clock. Instead, we have to do all the work on every clock.
+ * @param irq
```

Comment:
can we not have these littering the comments? if you want to comment on parameters, do it in the declaration, that's why I like to keep each parameter on one line...